### ISSUE TITLE
self.log validation_step Misconfiguration error

### ISSUE THREAD

**Issue by rongcuid** [PyTorchLightning/pytorch-lightning#8918]
## 🐛 Bug

After an update from 1.3.0 to 1.4.2, existing code breaks with:

```
  File "/home/rongcuid/miniconda3/envs/pytorch/lib/python3.8/site-packages/pytorch_lightning/trainer/connectors/logger_connector/result.py", line 464, in log
    raise MisconfigurationException(
pytorch_lightning.utilities.exceptions.MisconfigurationException: You called `self.log(loss/validate, ...)` twice in `validation_step` with different arguments. This is not allowed
```
There is only one `self.log` statement in my validation loop. However, there is an identical counterpart in the training step, which uses "loss/train" as name. It worked in 1.3.

### Expected behavior

Logs validation loss correctly.

### Environment

```
* CUDA:
	- GPU:
		- NVIDIA GeForce GTX 1080
	- available:         True
	- version:           10.2
* Packages:
	- numpy:             1.21.1
	- pyTorch_debug:     False
	- pyTorch_version:   1.9.0
	- pytorch-lightning: 1.4.2
	- tqdm:              4.62.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         x86_64
	- python:            3.8.11
	- version:           #21~1626191760~20.04~55de9c3-Ubuntu SMP Wed Jul 21 20:31:55 UTC
```

### Additional context

**Comment by rongcuid** [comment]
As a reference, my logging statement is the following:

```
        self.log("loss/validate", {
            "total": loss.detach(),
            "mel": loss_mel.detach(),
            "spec": loss_lin.detach(),
            "done": loss_done.detach(),
        })
```

**Comment by carmocca** [comment]
Hi!

As a workaround, I believe setting `Trainer(num_sanity_val_steps=0)` will avoid the error.

Thank you for your patience!

**Comment by rongcuid** [comment]
The workaround would crash similarly on second validation epoch

**Comment by tchaton** [comment]
Dear @rongcuid, 

Thanks for finding this bug. It seems the meta object is being dropped. Investigating :)

Best,
T.C
